subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/DefaultResourceSnapshotterCacheService.java

```diff
@@ -57,4 +76,10 @@ private static HashCode resourceHashCacheKey(HashCode contentHash, HashCode conf
         hasher.putHash(contentHash);
         return hasher.hash();
     }
+
+    @FunctionalInterface
+    private interface HashCodeSupplier {
+        @Nullable
+        HashCode get() throws IOException;
+    }
```

Comment:
This can now be replaced with `IoSupplier<HashCode>`.

Reply:
\o/ Replaced.